### PR TITLE
Swift v1: add support for new object versioning

### LINKED ISSUE #1489: Swift V1: Add support for objects versioning
### Affected Resource(s)

- openstack_objectstorage_container_v1

### Expected Behavior
Currently the object storage resource supports only legacy versioning support. A new versioning was introduced in Swift https://docs.openstack.org/swift/latest/middleware.html#module-swift.common.middleware.versioned_writes.object_versioning and it not yet supported in this provider.

### Actual Behavior

Currently only legacy versioning is supported. New versioning implementation has better compatibility with S3 API emulation, and there should be a way to trigger this option.

### References
- https://github.com/gophercloud/gophercloud/pull/2571

### PR DESCRIPTION
**Pull request by kayrus**
Fixes #1489

### CODE REVIEW

**Comment by kayrus** [comment]
@ozerovandrei please review.

File: openstack/resource_openstack_objectstorage_container_v1.go [hunk 1]
```diff
@@ -26,6 +26,15 @@ func resourceObjectStorageContainerV1() *schema.Resource {
 			StateContext: schema.ImportStatePassthroughContext,
 		},
 
+		SchemaVersion: 1,
+		StateUpgraders: []schema.StateUpgrader{
```
Comment (nikParasyr) [inline openstack/resource_openstack_objectstorage_container_v1.go:30]: This is interesting :). Think its the first time we use stateUpgrader

File: .github/workflows/functional-objectstorage.yml [hunk 2]
```diff
@@ -44,6 +44,9 @@ jobs:
           conf_overrides: |
             SWIFT_ENABLE_TEMPURLS=True
             SWIFT_TEMPURL_KEY=secretkey
+            [[post-config|\$SWIFT_CONFIG_PROXY_SERVER]]
```
Comment (nikParasyr) [inline .github/workflows/functional-objectstorage.yml:47]: are you sure this is loaded as you expect it to be?

**Review by nikParasyr — COMMENTED** [review]
This is still a breaking change right? 
The stateUpgrader will "move" what was on `versioning` to versioning_legacy but the user will still have to update his resource definition.
Or did i misunderstood something?

Moreover for some reason the tests are being skipped:
```
2023-02-28T20:52:42.1952225Z === RUN   TestAccObjectStorageV1Container_versioning
2023-02-28T20:52:42.1953883Z 2023-02-28T20:52:42.195Z [DEBUG] sdk.helper_resource: Starting TestCase: test_name=TestAccObjectStorageV1Container_versioning
2023-02-28T20:52:42.1954805Z 2023-02-28T20:52:42.195Z [DEBUG] sdk.helper_resource: Calling TestCase PreCheck: test_name=TestAccObjectStorageV1Container_versioning
2023-02-28T20:52:42.1955825Z     provider_test.go:270: this is not supported below stable/ussuri, testing in 
2023-02-28T20:52:42.1956481Z --- SKIP: TestAccObjectStorageV1Container_versioning (0.00s)
2023-02-28T20:52:42.1958238Z PASS
```

Ill try to find some time to see why

File: .github/workflows/functional-objectstorage.yml (hunk 2 above)
Comment (kayrus) [reply .github/workflows/functional-objectstorage.yml:47]: yes. see https://github.com/gophercloud/gophercloud/pull/2573

**Comment by kayrus** [comment]
> This is still a breaking change right? The stateUpgrader will "move" what was on versioning to versioning_legacy but the user will still have to update his resource definition.

correct. I think this is the most optimal solution.

> Moreover for some reason the tests are being skipped:

hm... you're right... `testAccSkipReleasesBelow(t, "stable/ussuri")` doesn't work as intended.

**Review by nikParasyr — APPROVED** [review]
Ok. tests seem to be running now and passing:
```
2023-03-01T07:25:46.8268716Z 2023-03-01T07:25:46.820Z [DEBUG] sdk.helper_resource: Called TestCase CheckDestroy: test_name=TestAccObjectStorageV1Container_versioning test_terraform_path=/tmp/plugintest-terraform3968591059/terraform test_working_directory=/tmp/plugintest195555014 test_step_number=1
2023-03-01T07:25:46.8269514Z 2023-03-01T07:25:46.820Z [DEBUG] sdk.helper_resource: Finished TestCase: test_name=TestAccObjectStorageV1Container_versioning
2023-03-01T07:25:46.8333324Z --- PASS: TestAccObjectStorageV1Container_versioning (3.15s)
```
I'll create an issue to have a look at why the `testAccSkipReleasesBelow` was not working as expected.

LGTM as long as we are ok with the breaking change (i do think as well thats it's the best way forward). Would like @ozerovandrei opinion as well

**Review by ozerovandrei — APPROVED** [review]
LGTM

**Pull request merged**